source/byte_buf.c

```diff
@@ -53,26 +66,17 @@ int aws_byte_buf_init_copy(struct aws_byte_buf *dest, struct aws_allocator *allo
 }
 
 bool aws_byte_buf_is_valid(const struct aws_byte_buf *const buf) {
```

Comment:
The new invariant assumes an empty `aws_byte_buf` is valid, any comments @justinboswell @bretambrose @danielsn ?